### ISSUE TITLE
VRPLIB v2

### ISSUE THREAD

**Issue by leonlan** [PyVRP/VRPLIB#114]
This issue keeps track of plans for a new major release of VRPLIB. This is a long-term goal and will be developed in parallel with the needs of PyVRP.

`vrplib` (v1) was originally designed to support any kind of free-form VRPLIB instance, mainly because the VRPLIB format was not well-defined. Its main use cases were reading the instances for the CVRP X-instances, the VRPTW EURO-NeurIPS instances and some of the LKH-3 instances.

PyVRP is continuously adding support for [new VRP variants](https://github.com/PyVRP/PyVRP/issues/441) and we are extending the [VRPLIB format](https://pyvrp.org/dev/supported_vrplib_fields.html) to support this. At some point, we will have a large collection of VRPLIB instances and a better well-defined format. 

----

In view of these developments, I propose the following todos for a next major release:

- [ ] A well-defined VRPLIB format for many VRP variants. Basically https://pyvrp.org/dev/supported_vrplib_fields.html combined with the current README. 
	- This VRPLIB format definition can be version-controlled and could evolve together with the `vrplib` package.
- [ ] Reading instances and solutions should return a `Instance` or `Solution` object with well-defined VRPLIB attributes.
	- [ ] It would be nice to keep a flexible way of reading VRPLIB instances with undefined specifications/sections, so that we can easily extend the VRPLIB format.
- [ ] Maintain a library of VRPLIB instances
	- We already do this in PyVRP/Instances, but we only keep selected instances for benchmarking. There are many more VRPLIB instances that I have created that we don't have a place for anywhere. It would be nice to have these stored somewhere as well for others to use.
- [ ] Drop support for downloading instances. This is a legacy feature that originated from when this package was called `cvrplib`.

**Comment by N-Wouda** [comment]
> [ ] Read/write should be a no-op, meaning that when you read an instance and then write the instance, then this give precisely the same instance again.

I was going to write this out into a separate issue, but since it's already tracked here I won't have to :). It'll be good to have this in because it makes adapting instances to VRPLIB much easier.

**Comment by leonlan** [comment]
I'm going to check out the [Lark](https://github.com/lark-parser/lark) parsing library. If we use the LALR(1) parsing algorithm, it will also [generate a standalone parser](https://github.com/lark-parser/lark/blob/master/docs/tools.md#stand-alone-parser) which is very neat and then Lark is only a development dependency.

**Comment by leonlan** [comment]
We are no longer working towards the plans listed in this issue in the next major release. Instead, we will remove the download functionality and bump to the next major version.

Moving forward, we'll maintain `vrplib` in its current state without further interface modifications. While the existing implementation isn't ideal, it serves its original purpose well by providing easy reading functionality to academic benchmark instances.
